### ISSUE TITLE
Represent date-times in long ISO format

### ISSUE THREAD

**Issue by MetRonnie** [cylc/cylc-ui#962]
**Describe exactly what you would like to see in an upcoming release**

We should consider displaying date-times in the long ISO format in the UI, even if the cycle point format of a workflow is the short format, as it will make cycle points easier to read. I.e. `2000-01-01T00:00Z` instead of `20000101T0000Z`.

**Pull requests welcome!**

**Comment by hjoliver** [comment]
Agreed, that would be nice. Although it might be a little dangerous, if it leads users to use the wrong datetime format on the command line? (maybe we can automatically convert to the workflow format... I think we do that already for `cylc suite-state` xtriggers).

**Comment by MetRonnie** [comment]
I don't think it matters what users use on the command line? According to [the docs](https://cylc.github.io/cylc-doc/nightly/html/reference/config/workflow.html#flow.cylc[scheduler]cycle%20point%20format)
> The time zone you specify here will be used only for writing/dumping cycle points

**Comment by hjoliver** [comment]
> I don't think it matters what users use on the command line? According to [the docs](https://cylc.github.io/cylc-doc/nightly/html/reference/config/workflow.html#flow.cylc%5Bscheduler%5Dcycle%20point%20format)
> 
> > The time zone you specify here will be used only for writing/dumping cycle points

I mean the whole datetime format, not just the timezone.

E.g. for a monthly-cycling workflow with this:
```
cycle point format = %Y-%m
```

The CLI has to use the same format as the scheduler because no conversion is done. If `wf//2025-01/foo` is running:
```
$ cylc kill wf//2025-01/foo  # OK
$ cylc kill/wf//202501/foo  # ERROR (task not found)
```

**Comment by MetRonnie** [comment]
Argh, clearly I needed more coffee yesterday ☕

**Comment by MetRonnie** [comment]
Since https://github.com/cylc/cylc-flow/pull/4881 the CLI no longer has to be given the same format as the workflow cycle point format

**Comment by MetRonnie** [comment]
Just had a thought about this the other day. Potentially we could do a similar thing to i18n and have a function we call on every cycle point displayed in all vue templates, like https://github.com/cylc/cylc-ui/blob/afb15003f26fa3e11225a828da06cf28a3219d6a/src/components/cylc/workspace/Toolbar.vue#L155-L157
